File: react_web/src/services/medicalService.js
import api from './api';

// POST /clubs/:clubId/teams/:teamId/players/:playerUserId/medical
export const createMedicalRecord = async (clubId, teamId, playerUserId, request) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/medical`, request);
  return data;
};

// PUT /clubs/:clubId/teams/:teamId/medical/:recordId
export const updateMedicalRecord = async (clubId, teamId, recordId, request) => {
  const { data } = await api.put(`/clubs/${clubId}/teams/${teamId}/medical/${recordId}`, request);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/players/:playerUserId/medical
export const getPlayerMedicalRecords = async (clubId, teamId, playerUserId) => {
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/medical`);
  return data;
};

// PATCH /clubs/:clubId/teams/:teamId/medical/:recordId/clearance
export const updateMedicalClearance = async (clubId, teamId, recordId, request) => {
  const { data } = await api.patch(`/clubs/${clubId}/teams/${teamId}/medical/${recordId}/clearance`, request);
  return data;
};

// POST /clubs/:clubId/teams/:teamId/medical/:recordId/document-requests
export const requestMedicalDocument = async (clubId, teamId, recordId, request) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/medical/${recordId}/document-requests`, request);
  return data;
};

// POST /players/me/medical/document-requests/:requestId/upload
export const uploadMedicalDocument = async (requestId, file) => {
  const formData = new FormData();
  formData.append('file', file);
  const { data } = await api.post(`/players/me/medical/document-requests/${requestId}/upload`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return data;
};

// GET /medical/document-requests/:requestId/download
export const downloadMedicalDocument = async (requestId, fallbackName = 'medical-document') => {
  const response = await api.get(`/medical/document-requests/${requestId}/download`, {
    responseType: 'blob',
  });
  const url = window.URL.createObjectURL(response.data);
  const link = document.createElement('a');
  const disposition = response.headers['content-disposition'] || '';
  const match = disposition.match(/filename\*=UTF-8''([^;]+)|filename="?([^"]+)"?/i);
  link.href = url;
  link.download = decodeURIComponent(match?.[1] || match?.[2] || fallbackName);
  document.body.appendChild(link);
  link.click();
  link.remove();
  window.URL.revokeObjectURL(url);
};

// GET /players/me/medical
export const getMyMedicalRecords = async () => {
  const { data } = await api.get('/players/me/medical');
  return data;
};

File: eq-master/lib/services/medical_service.dart
import 'dart:typed_data';

import '../models/api_models.dart';
import 'api_client.dart';

class MedicalService {
  final ApiClient _api = ApiClient.instance;

  Future<List<MedicalRecordDto>> getPlayerMedical(
      String clubId, String teamId, String playerId) async {
    final json =
        await _api.get('/clubs/$clubId/teams/$teamId/players/$playerId/medical');
    return (json as List)
        .map((e) =>
            MedicalRecordDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<MedicalRecordDto> createMedicalRecord(
      String clubId, String teamId, String playerId, Map<String, dynamic> body) async {
    final json = await _api.post(
      '/clubs/$clubId/teams/$teamId/players/$playerId/medical',
      body: body,
    );
    return MedicalRecordDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<MedicalRecordDto> updateMedicalRecord(
      String clubId, String teamId, String recordId, Map<String, dynamic> body) async {
    final json = await _api.put(
      '/clubs/$clubId/teams/$teamId/medical/$recordId',
      body: body,
    );
    return MedicalRecordDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<void> updateClearance(
      String clubId, String teamId, String recordId, bool cleared) async {
    await _api.patch(
      '/clubs/$clubId/teams/$teamId/medical/$recordId/clearance',
      body: {'isCleared': cleared},
    );
  }

  Future<MedicalDocumentRequestDto> requestDocument(
      String clubId, String teamId, String recordId, Map<String, dynamic> body) async {
    final json = await _api.post(
      '/clubs/$clubId/teams/$teamId/medical/$recordId/document-requests',
      body: body,
    );
    return MedicalDocumentRequestDto.fromJson(
      Map<String, dynamic>.from(json as Map),
    );
  }

  Future<void> deleteMedicalRecord(
      String clubId, String teamId, String recordId) async {
    await _api.post('/clubs/$clubId/teams/$teamId/medical/$recordId/delete');
  }

  Future<dynamic> uploadDocument(
      String requestId, String filePath, String fileName) {
    return _api.uploadFile(
      '/players/me/medical/document-requests/$requestId/upload',
      fileField: 'file',
      filePath: filePath,
      fileName: fileName,
    );
  }

  /// Downloads the uploaded document bytes (authenticated).
  /// Returns a record with the bytes, content type, and file name.
  Future<({Uint8List bytes, String contentType, String fileName})>
      downloadDocument(String requestId) async {
    final response = await _api
        .getFile('/medical/document-requests/$requestId/download');
    final contentType =
        response.headers['content-type'] ?? 'application/octet-stream';
    // Try to extract filename from content-disposition header
    var fileName = 'document';
    final disposition = response.headers['content-disposition'];
    if (disposition != null) {
      final pattern = RegExp('filename\\*?=["\\x27]?([^"\\x27;\\r\\n]+)');
      final match = pattern.firstMatch(disposition);
      if (match != null) fileName = match.group(1)!;
    }
    return (
      bytes: response.bodyBytes,
      contentType: contentType,
      fileName: fileName,
    );
  }
}
